### PR TITLE
Migrate to prosemirror

### PR DESCRIPTION
**Pull request by andyk**
Large rewrite of most of webapp (thankfully it's small!)

Migrate to using a prosemirror Editor instead of a list of `<textarea>`s.

Did this for a more intuitive thought editor UX (feels like editing a textfile, one thoughts are delimited by paragraph breaks, i.e. pressing enter makes new paragraph which is a new thought).

Also migrates us from agents table to thoughts table (where agent_name is a column)

Highlight text added by LLM in Green. Text typed by user has no highlight (highlighting still not persisted to database so lost at reload)

Also show GPT4 and HF model in dropdown (if HF env vars are set)

Not all of the intuitive keyboard bindings work yet (highlighting and deleting across thoughts, i.e. rows doesn't work)

TODO: 
* allow user to select agent_name (currently defaults to "gimli" and can't be changed
* Persist highlighting
* improve keybindings, maybe override Ctl+A to be just current Thougth (i.e. paragraph)?
* Persist the prompt used to generate any given LLM call (make available via hover or right click menu?)

### CODE REVIEW

File: src/App.tsx [hunk 1]
```diff
@@ -1,45 +1,87 @@
...
+import { Database } from "./database.types";
+
+type Thought = Database["public"]["Tables"]["thoughts"]["Row"];
+
+const removeHighlightOnInputPlugin = new Plugin({
+  appendTransaction(transactions, oldState, newState) {
+    let newTransaction: Transaction | null = null;
+    let markAdded = false;
+    transactions.forEach((tr) => {
+      console.log("Inside appendTransaction, handling tr: ", tr);
```
Comment (recarroll) [inline src/App.tsx:29]: Can we cut down on some of the logging in this function? Lots of noise in the console

File: src/App.tsx [hunk 2]
```diff
@@ -1,45 +1,87 @@
...
+              }
+            }
+          }
+        });
+      }
+    });
+
+    // Only append transactions if we've actually created any
+    if (newTransaction !== null && !markAdded) {
+      console.log("returning newTransaction: ", newTransaction);
```
Comment (recarroll) [inline src/App.tsx]: Here too

File: src/App.tsx [hunk 3]
```diff
@@ -116,339 +157,367 @@ function App() {
...
-      setAgent((old) => {
-        if (old === null) {
-          return old;
-        }
-        const newThoughts = [...old.thoughts];
-        newThoughts.splice(index + 1, 0, {
-          body: EditorState.createEmpty(),
-          human_generated_slices: [],
+        state.doc.descendants((node, pos) => {
+          console.log("Looking at node: ", node.attrs.index, node.attrs.id, pos);
```
Comment (recarroll) [inline src/App.tsx]: More

**Review by justin-fiedler — APPROVED** [review]
Awesome. The new editor is great. 

I noticed a bug where a thought will fail to write to DB if you create another new thought before the debounce time.

**Pull request merged**